site/app/libraries/FileUtils.php

```diff
@@ -569,14 +569,24 @@ public static function validateUploadedFiles(array $files): array {
         $validator = function ($file) {
             $max_size = Utils::returnBytes(ini_get('upload_max_filesize'));
             $name = $file['name'];
+
+            //if the error flag is set for this file, no guarantee tmp_name points to a valid file, exit early
+            if ($file['error'] !== UPLOAD_ERR_OK) {
+                return [
+                    'name' => $name,
+                    'type' => 'Unknown',
```

Comment:
Instead of a magic string value which isn't a valid MIME type, what do you think about using null for the error case instead?

Reply:
My expectation is when the UPLLOAD_ERR_OK flag is false, basically the caller will never use the result and just propogate the error forwards.

Looking through the existing uses (just eyeballing) that seems like the case, I'd personally avoid using null just in the off chance someone does try to access the value and doesn't perform a null check because the expectation from the happy path case is this value is always a string.

But thats just my opinion i don't like using null in php due to the weak php type checking so i have no idea if a null is ever actually supposed to be null or its something weird happened.

Not a strong opinion however, happy to change to a null if you'd like to standardize how handle error handling :)

Reply:
Do agree that using an unofficial mime type might cause problems, I could see us using "application/octet-stream" to preserve using a string type, this apparently means just a stream of bytes according to https://stackoverflow.com/questions/1176022/unknown-file-type-mime

Reply:
Decided to go with null because operating on an unknown mime type would probably have the same effects of null anyways even if the type is same